raiden-ts/src/raiden.ts

```diff
@@ -1129,6 +1129,54 @@ export class Raiden {
     return receipt.transactionHash as Hash;
   }
 
+  /**
+   * Registers and creates a new token network for the provided token address.
+   * Throws an error, if
+   * <ol>
+   *  <li>Executed on main net</li>
+   *  <li>`token` is not a valid address</li>
+   *  <li>Token is already registered</li>
+   *  <li>Token could not be registered</li>
+   * </ol>
+   *
+   * @param token - Address of the token to be registered
+   * @param channelParticipantDepositLimit - The deposit limit per channel participant
+   * @param tokenNetworkDepositLimit - The deposit limit of the whole token network
+   * @returns Address of new token network
+   */
+  public async registerToken(
+    token: string,
+    channelParticipantDepositLimit: BigNumberish = MaxUint256,
+    tokenNetworkDepositLimit: BigNumberish = MaxUint256,
+  ): Promise<Address> {
+    // Check whether address is valid
+    assert(Address.is(token), [ErrorCodes.DTA_INVALID_ADDRESS, { token }], this.log.info);
+
+    // Check whether we are on a test network
+    assert(this.deps.network.chainId !== 1, ErrorCodes.RDN_REGISTER_TOKEN_MAINNET, this.log.info);
+
+    const { signer } = chooseOnchainAccount(this.deps, this.config.subkey);
+
+    const registryAddress = this.deps.contractsInfo.TokenNetworkRegistry.address;
+    const tokenNetworkRegistry = TokenNetworkRegistry__factory.connect(registryAddress, signer);
+
+    // Check whether token is already registered
+    const isRegistered = token in this.state.tokens;
+    assert(!isRegistered, ErrorCodes.RDN_REGISTER_TOKEN_REGISTERED, this.log.info);
+
+    const receipt = await callAndWaitMined(
+      tokenNetworkRegistry,
+      'createERC20TokenNetwork',
+      [token, channelParticipantDepositLimit, tokenNetworkDepositLimit],
+      ErrorCodes.RDN_REGISTER_TOKEN_FAILED,
+      { log: this.log },
+    );
+    await waitConfirmation(receipt, this.deps);
+
+    await this.monitorToken(token);
+    return receipt.contractAddress as Address;
```

Comment:
I think this will probably return the contract's address where the tx was executed, in this case, TokenNetworkRegistry, which may be misleading. You probably want to:
```suggestion
    return await this.monitorToken(token);
```